### PR TITLE
Run first scenario

### PR DESCRIPTION
**Pull request by anciuvaga**
- Moved .gitignore to root folder
- Refactored used classes
- Added xpaths for first scenario
- Generated methods
- Run first scenario
image

### CODE REVIEW

File: src/main/java/hooks/Hooks.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import managers.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+
+public class Hooks {
+    private ConfigFileReader fileReader;
+    private WebDriverManager webDriverManager;
+
+    private static WebDriver driver;
+
+    public Hooks() {}
```
Comment (CristianManic) [inline src/main/java/hooks/Hooks.java:15]: do you really need it?

File: src/main/java/runner/TestRunner.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        monochrome = true
+        )
+public class TestRunner {
+
+}
+
+
+
+
+
```
Comment (CristianManic) [inline src/main/java/runner/TestRunner.java:21]: remove all blank spaces

File: src/test/java/stepDefinition/TestAndreiCiuvagaSteps.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import pages.HomePage;
+import pages.PageObject;
+
+import java.util.List;
+
+public class TestAndreiCiuvagaSteps extends PageObject {
```
Comment (CristianManic) [inline src/test/java/stepDefinition/TestAndreiCiuvagaSteps.java:13]: 1. this class is not a page object
2. class name should represent business functionality, you name is not related to business

File: src/test/resources/features/AndreiCiuvaga.feature [hunk 4]
```diff
@@ -0,0 +1,40 @@
+@andreiciuvaga
+Feature:Andrei's Feature
```
Comment (CristianManic) [inline src/test/resources/features/AndreiCiuvaga.feature:2]: rename feature and feature files related to business

Comment (anciuvaga) [reply src/test/resources/features/AndreiCiuvaga.feature:2]: Ok, thanks

File: src/test/java/stepDefinition/TestAndreiCiuvagaSteps.java (hunk 3 above)
Comment (anciuvaga) [reply src/test/java/stepDefinition/TestAndreiCiuvagaSteps.java:13]: Ok, I am going to rename it

File: src/main/java/hooks/Hooks.java (hunk 1 above)
Comment (anciuvaga) [reply src/main/java/hooks/Hooks.java:15]: Yes, there is no need for the constructor here, thanks

File: src/test/java/stepDefinition/TestAndreiCiuvagaSteps.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    @Given("^User is on 'HomePage'$")
+    public void userIsOnHomePage() {
+    }
```
Comment (amiordan) [inline src/test/java/stepDefinition/TestAndreiCiuvagaSteps.java:16]: Why is this step empty ?

Comment (anciuvaga) [reply src/test/java/stepDefinition/TestAndreiCiuvagaSteps.java:16]: Hi!
We use @before for every scenario so when we execute this piece of code we are get landed on Homepage, so I put Given in Background as is common for every scenario, we not doing any action here, so I left it empty

File: src/main/java/pages/PageObject.java [hunk 6]
```diff
@@ -1,15 +1,39 @@
...
+    private HomePage homePage;
+    private ProductPage productPage;
+    private ShoppingCart shoppingCart;
+    private ProductDetailsPage productDetailsPage;
 
-    PageObject(WebDriver driver) {
-        this.driver = driver;
+    public PageObject() {
         PageFactory.initElements(driver, this);
     }
```
Comment (CristianManic) [inline src/main/java/pages/PageObject.java:17]: revert this change, Page object constructor must take as input parameter the webdriver

File: src/main/java/pages/PageObject.java [hunk 7]
```diff
@@ -1,15 +1,39 @@
...
 import org.openqa.selenium.support.PageFactory;
 
 public abstract class PageObject {
 
-    protected WebDriver driver;
+    private WebDriver driver = Hooks.getDriver();
+    private HomePage homePage;
+    private ProductPage productPage;
+    private ShoppingCart shoppingCart;
+    private ProductDetailsPage productDetailsPage;
```
Comment (CristianManic) [inline src/main/java/pages/PageObject.java:13]: revert this, PageObject class is an abstract class that must contain just common staff for all page objects

File: src/main/java/pages/PageObject.java [hunk 8]
```diff
@@ -1,15 +1,39 @@
...
+        return (productPage == null) ? new ProductPage() : productPage;
+    }
+
+    public ShoppingCart getShoppingCart(){
+        return (shoppingCart == null) ? new ShoppingCart() : shoppingCart;
+    }
+
+    public ProductDetailsPage getProductDetailsPage(){
+        return (productDetailsPage == null) ? new ProductDetailsPage() : productDetailsPage;
+    }
```
Comment (CristianManic) [inline src/main/java/pages/PageObject.java:37]: this methods are not related to this class

File: src/main/java/pages/ProductDetailsPage.java [hunk 9]
```diff
@@ -1,10 +1,62 @@
...
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import static org.junit.Assert.assertTrue;
 
 public class ProductDetailsPage extends PageObject {
 
-    ProductDetailsPage(final WebDriver driver) {
-        super(driver);
+    public ProductDetailsPage() {}
```
Comment (CristianManic) [inline src/main/java/pages/ProductDetailsPage.java:13]: revert this change

File: src/main/java/pages/ProductDetailsPage.java [hunk 10]
```diff
@@ -1,10 +1,62 @@
...
+        nameField.sendKeys(name);
+        phoneField.sendKeys(phone);
+        submitFormComandaImediat.click();
+    }
+
+    public void validateSuccessMessage() {
+        WebDriverWait wait = new WebDriverWait(getDriver(), 30);
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//div[@class='product-buy-success']")));
+        assertTrue(successMessage.isDisplayed());
     }
```
Comment (CristianManic) [inline src/main/java/pages/ProductDetailsPage.java:61]: all this methods are not related to this class
